src/Microsoft.Graph.Core/Extensions/BaseRequestExtensions.cs

```diff
@@ -107,6 +107,29 @@ public static class BaseRequestExtensions
             return baseRequest;
         }
 
+        /// <summary>
+        /// Sets the maximum time for request retries to the default Retry Middleware Handler for this request.
+        /// This only works with the default Retry Middleware Handler.
+        /// If you use a custom Retry Middleware Handler, you have to handle it's retrieval in your implementation.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="baseRequest">The <see cref="BaseRequest"/> for the request.</param>
+        /// <param name="retriesTimeLimit">The retriestimelimit for the request.</param>
+        /// <returns></returns>
+        public static T WithRetriesTimeLimit<T>(this T baseRequest, int retriesTimeLimit) where T : IBaseRequest
```

Comment:
I suggest we make this an overload of WithMaxRetry and use Timespan as the type.  Using int as the datatype makes it unclear when reading code if this is milliseconds, seconds, minutes.

Reply:
I could make it an overload method of `WithMaxRetry`. 
With regards to the datatype, I chose to use `int` so as to maintain consistency within the project; The `Delay` parameter is `int` typed as well. I could probably add more descriptive comments in the xml documentation.

Reply:
Being consistent with delay make sense considering Delay() is public.  But I think that Delay() should not be public.  @peombwa Any idea why we made Delay public?  I think anyone looking at the surface API of this class would question why we chose int.  I definitely think we should make retriesTimeLimit a TimeSpan.

Reply:
I have chosen to go with TimeSpan :)

Reply:
It was inherited code, but since we've already changed its interface in this PR, we can go ahead and change it to internal (our tests depends on it). I'm all for TimeSpan as opposed to int.

Reply:
Changing the `Delay()` to TimeSpan might lead to breaking changes as we pass the `Delay()` value as an `int` to this delegate: 
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/2c23a8bdeeaa608e7e5e36494b443e5982fbef6b/src/Microsoft.Graph.Core/Requests/Middleware/Options/RetryHandlerOption.cs#L79

Reply:
It would be a breaking change to change Delay() to use TimeSpan, but then so is making it internal. It wasn't intended to be public so this is really a bug fix. Go ahead and make this change.